docs/source/shared/top-level-await.mdx

```diff
@@ -0,0 +1 @@
+> In the examples below, we use top-level [`await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) calls to start our server asynchronously. Check out our [Getting Started](/apollo-server/getting-started#step-2-install-dependencies) guide to see how we configured our project to support this.
```

Comment:
I think linking to `async` function docs isn't quite helpful, since top-level `await` actually allows us to drop our previously-required `async` functions.

How about this link?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await

We could also mention that support for top-level `await` in ES modules has been available in Node.js (unflagged) since v14.8.0.
https://nodejs.org/en/blog/release/v14.8.0/

Reply:
Oh, great point! I used the link you suggested. In the interest of keeping this note short, I feel like we can omit when Node started supporting top-level `await` for now (especially because they do have the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#browser_compatibility) at the end of the page we are now linking to!)